### ISSUE TITLE
Poor ONNX parity when converting (siamese) models

### ISSUE THREAD

**Issue by adewynter** [microsoft/onnxruntime#13633]
### Describe the issue

We are trying to convert a siamese PyTorch model to ONNX and perform inference with it on a CPU instance. However:

1. We are observing that our model does not meet neither the speed expectation nor the accuracy bar when running the ONNX model using ORT for inference.
2. Our conversion script hangs when using an F48v2 instance. It works fine in GPU, although issue (1) above still appears.

A full description and reproducible steps are in our [ONNX ticket](https://github.com/microsoft/onnx-converters-private/issues/108). It also includes an analysis from the ONNX team on the converted graph, where they show that the output of the export algorithm has disparities with the original model.

It might be worthwhile to note that:
1. A siamese architecture calls the same module twice, as opposed to maintaining two copies of the module. Not sure whether it is helpful or not!
2. We have extra postprocessing steps done, so the error found in the ticket appears to be amplified. More on this on the next section.

### To reproduce

A full description and reproducible steps are in our [ONNX ticket](https://github.com/microsoft/onnx-converters-private/issues/108). It also includes an analysis from the ONNX team on the converted graph.

### Platform

Linux

### OS Version

Ubuntu

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

I've tried everything between 13.1 and 11.0

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU

**Comment by tianleiwu** [comment]
@adewynter, 

In the ONNX issue, the parity seems good to me:
> precision for CUDA up to atol=1e-6 and rtol=1e-3
> precision for CPU up to atol=1e-6 and rtol=3e-2
Is this issue about performance? For performance issue, do you have script to reproduce? 

Generally, a few things to try for performance: (1) use a few inference runs to warm up before starting measurement (2) use IO Binding for CUDA (3) Try [transformer optimization tool](https://github.com/microsoft/onnxruntime/tree/main/onnxruntime/python/tools/transformers#readme) to optimize the model.

**Comment by tianleiwu** [comment]
Accuracy evaluation after disable ORT graph optimization:
* transformers 4.20: {'accuracy_onnx': 0.0756459628464328, 'accuracy_torch': 0.08118009236756875}
* transformers 4.21: {'accuracy_onnx': 0.07937983336405027, 'accuracy_torch': 0.08118009236756875}
There is significant drop in accuracy. I suspect that it is related to https://github.com/microsoft/onnxruntime/issues/12843.

**Comment by adewynter** [comment]
That did it! Thank you so much for this! Additionally, onnxruntime is faster than the stock version, so it is a win-win :)

**Comment by thiagocrepaldi** [comment]
> After some analysis, the cause of accuracy loss is the position of attention_mask and token_type_ids is exchanged in the graph. Fixing it could recover the accuracy.

Out of curiosity, how did you changed the attention_mask and token_type_ids in the onnx graph?

**Comment by adewynter** [comment]
Turns out that that there was some discordance in the API: specifying the input names, as a list, and feeding in a dictionary as the inputs, will mean that the names would be unordered. Super weird.
